### ISSUE TITLE
Better handling of broken SPF records

### ISSUE THREAD

**Issue by schack** [domainaware/checkdmarc#19]
The domain bookmate.com has a broken SPF record. 
They forgot a mechanism in front of the ip address in one of the includes.
checkmark crashes on that.

**Comment by seanthegeek** [comment]
Fixed in `1.7.9`:

```json
{
  "domain": "bookmate.com",
  "base_domain": "bookmate.com",
  "mx": {
    "hosts": [
      {
        "preference": 10,
        "hostname": "aspmx.l.google.com",
        "addresses": [
          "173.194.196.27",
          "2607:f8b0:4001:c1d::1b"
        ]
      },
      {
        "preference": 20,
        "hostname": "alt2.aspmx.l.google.com",
        "addresses": [
          "74.125.141.26",
          "2607:f8b0:400c:c06::1b"
        ]
      },
      {
        "preference": 20,
        "hostname": "alt1.aspmx.l.google.com",
        "addresses": [
          "209.85.232.26",
          "2607:f8b0:400d:c03::1b"
        ]
      },
      {
        "preference": 30,
        "hostname": "aspmx5.googlemail.com",
        "addresses": [
          "209.85.202.26",
          "2a00:1450:400b:c03::1b"
        ]
      },
      {
        "preference": 30,
        "hostname": "aspmx3.googlemail.com",
        "addresses": [
          "74.125.141.26",
          "2607:f8b0:400c:c06::1a"
        ]
      },
      {
        "preference": 30,
        "hostname": "aspmx2.googlemail.com",
        "addresses": [
          "209.85.201.26",
          "2607:f8b0:400d:c0d::1a"
        ]
      },
      {
        "preference": 30,
        "hostname": "aspmx4.googlemail.com",
        "addresses": [
          "64.233.186.26",
          "2800:3f0:4003:c01::1b"
        ]
      }
    ],
    "warnings": []
  },
  "spf": {
    "record": "v=spf1 include:_spf4.bookmate.com include:_spf6.bookmate.com include:_spf.google.com include:mail.zendesk.com include:mailgun.org include:sendgrid.net include:_spf.zdsys.com include:spf.braintreegateway.com -all",
    "valid": false,
    "error": "_spf6.bookmate.com: Expected mechanism at position 109 in: v=spf1 ip6:2a00:7c80:0:22::/64 ip6:2a01:4f8:b0:7158::/64 ip6:2a01:4f8:b0:a143::/64 ip6:2a01:4f8:b0:c0b6::/64 2a01:4f8:10a:2392::/64 -all"
  },
  "dmarc": {
    "record": "v=DMARC1;p=none;sp=none;rua=mailto:dmarc@bookmate.com;fo=1",
    "valid": true,
    "location": "bookmate.com",
    "tags": {
      "v": {
        "value": "DMARC1",
        "explicit": true
      },
      "p": {
        "value": "none",
        "explicit": true
      },
      "sp": {
        "value": "none",
        "explicit": true
      },
      "rua": {
        "value": [
          {
            "scheme": "mailto",
            "address": "dmarc@bookmate.com",
            "size_limit": null
          }
        ],
        "explicit": true
      },
      "fo": {
        "value": [
          "1"
        ],
        "explicit": true
      },
      "adkim": {
        "value": "r",
        "explicit": false
      },
      "aspf": {
        "value": "r",
        "explicit": false
      },
      "pct": {
        "value": 100,
        "explicit": false
      },
      "rf": {
        "value": [
          "afrf"
        ],
        "explicit": false
      },
      "ri": {
        "value": 86400,
        "explicit": false
      }
    },
    "warnings": []
  }
}
```

**Comment by schack** [comment]
Thanks, but it looks like the fix broke something else.
a domain like co3.dk now has an SPF error using checkdmarc, but not according to dmarcian.com's SPF validator

**Comment by seanthegeek** [comment]
@schack  Fixed for real in 1.7.10

**Comment by schack** [comment]
Thank you !
